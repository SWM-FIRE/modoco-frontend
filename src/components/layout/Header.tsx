import React, { useState } from 'react';
import styled from 'styled-components';
import { useNavigate } from 'react-router-dom';
import UserStore from '../../stores/userStore';
import Profile from './Profile';
import HeaderProfileModal from '../HeaderProfile/HeaderProfileModal';
import useSetSelf from '../../hooks/useSetSelf';

export default function Header() {
  const { isLogin } = UserStore();
  const [headerProfileModal, toggleHeaderProfileModal] =
    useState<boolean>(false);
  const navigate = useNavigate();

  useSetSelf();
  const clickLogo = () => {
    if (localStorage.getItem('access_token')) {
      navigate('/main');
    } else {
      navigate('/');
    }
  };

  const toggleModal = () => {
    toggleHeaderProfileModal(!headerProfileModal);
  };

  return (
    <>
      {headerProfileModal && <Screen onClick={toggleModal} />}
      <Container>
        <Logo onClick={clickLogo}>modoco</Logo>
        {isLogin ? (
          <Profile
            toggleModal={toggleModal}
            headerProfile={headerProfileModal}
          />
        ) : (
          <ProfileSkeleton />
        )}
        {headerProfileModal && <HeaderProfileModal toggleModal={toggleModal} />}
      </Container>
    </>
  );
}

const ProfileSkeleton = styled.div`
  width: 9.1rem;
  height: 4.8rem;
  border-radius: 3rem;
  background-color: rgba(255, 255, 255, 0.06);
`;

const Screen = styled.div`
  position: fixed;
  width: 100vw;
  height: 100vh;
  z-index: 1;
`;

const Logo = styled.h1`
  cursor: pointer;
  font-size: 3.2rem;
  font-family: IBMPlexSansKRRegular, Arial;
  color: white;
  font-weight: 700;
`;

const Container = styled.div`
  width: 100vw;
  height: 10rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: solid #2b2e41 0.1rem;
  padding: 0 4.4rem;
  position: relative;
  background-color: #18181b;
`;
