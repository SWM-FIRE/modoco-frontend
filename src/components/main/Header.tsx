import styled from 'styled-components';
import { useNavigate } from 'react-router-dom';
import UserStore from '../../stores/userStore';
import MyAvatar from '../../assets/avatar/MyAvatar';
import LoginModalStore from '../../stores/loginModalStore';
import LogoutModalStore from '../../stores/logoutModalStore';
import LogoutModal from '../logout/LogoutModal';
import { ReactComponent as TopArrow } from '../../assets/svg/topArrow.svg';
import { ReactComponent as BottomArrow } from '../../assets/svg/bottomArrow.svg';

export default function Header() {
  const { nickname, avatar } = UserStore((state) => state);

  const { openLoginModal } = LoginModalStore((state) => state);
  const { isOpenLogoutModal, toggleLogoutModal } = LogoutModalStore();
  const navigate = useNavigate();

  const clickLogo = () => {
    navigate('/#/');
  };

  return (
    <Container>
      <Logo onClick={clickLogo}>modoco</Logo>
      {nickname ? (
        <Profile onClick={toggleLogoutModal}>
          <AvatarContainer>
            <MyAvatar num={Number(avatar)} />
          </AvatarContainer>
          <SvgComponent>
            {isOpenLogoutModal ? <TopArrow /> : <BottomArrow />}
          </SvgComponent>
        </Profile>
      ) : (
        <Login onClick={openLoginModal}>로그인</Login>
      )}
      {isOpenLogoutModal && <LogoutModal />}
    </Container>
  );
}

const Login = styled.button`
  width: 9.1rem;
  height: 4.8rem;
  border: 2px solid #494e5b;
  border-radius: 0.8rem;
  cursor: pointer;
  color: #fcfcf9;
  font-size: 1.6rem;
  font-family: IBMPlexSansKRRegular;
  font-weight: 700;
`;

const Logo = styled.div`
  cursor: pointer;
  font-size: 3.2rem;
  font-family: IBMPlexSansKRRegular, Arial;
  color: white;
  font-weight: 700;
`;

const AvatarContainer = styled.div`
  height: 4rem;
  width: 4rem;
  margin-left: 0.3rem;
  svg {
    height: 100%;
    width: 100%;
  }
`;

const Profile = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: IBMPlexSansKRRegular;
  font-size: 1.5rem;
  color: white;
  border: 1px solid #4b5563;
  border-radius: 5rem;
  cursor: pointer;
`;

const SvgComponent = styled.div`
  margin: 1.9rem;
  display: flex;
  align-items: center;
`;

const Container = styled.div`
  width: 100vw;
  height: 10rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: solid #2b2e41 0.1rem;
  padding: 0 4.4rem;
  position: relative;
  background-color: #18181b;
`;
