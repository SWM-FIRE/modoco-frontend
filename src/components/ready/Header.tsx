import React from 'react';
import styled from 'styled-components';
import { useNavigate } from 'react-router-dom';
import { useCreateMediaStream } from '../../hooks/useCreateMediaStream';
import { ReactComponent as LeftArrow } from '../../assets/svg/arrow-left.svg';
import UserMediaStreamStore from '../../stores/room/userMediaStreamStore';

export default function Header() {
  const { stopMediaStream } = useCreateMediaStream();
  const { userMediaStream } = UserMediaStreamStore();
  const navigate = useNavigate();

  const backToMain = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    if (userMediaStream) {
      stopMediaStream();
    }
    navigate(`/`);
  };

  return (
    <Container>
      <LeftArrowPosition onClick={backToMain}>
        <LeftArrow />
      </LeftArrowPosition>
    </Container>
  );
}

const LeftArrowPosition = styled.button`
  cursor: pointer;
  left: 8rem;
`;

const Container = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  height: 10rem;
  width: 90%;
`;
