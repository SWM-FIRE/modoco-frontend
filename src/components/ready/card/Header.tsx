import styled from 'styled-components';
import MyAvatar from '../../../assets/avatar/MyAvatar';

export default function Header({ data }) {
  return (
    <Container data-cy="ready-card-header">
      <AvatarControl>
        <MyAvatar num={Number(data?.moderator.avatar)} />
      </AvatarControl>
      <NameTag data-cy="ready-card-name">
        <Moderator isModerator={1}>방장</Moderator>
        <Moderator isModerator={0}>{data?.moderator.nickname}</Moderator>
      </NameTag>
    </Container>
  );
}

const Container = styled.div`
  display: flex;
  align-items: center;
  gap: 1.6rem;
  height: 4.4rem;
`;

const NameTag = styled.div`
  font-family: IBMPlexMonoRegular;
  font-size: 1.2rem;
  display: flex;
  gap: 0.2rem;
`;

const AvatarControl = styled.div`
  width: 4.4rem;
  svg {
    width: 100%;
  }
`;

const Moderator = styled.p<{ isModerator: number }>`
  color: ${(props) =>
    props.isModerator ? 'rgba(255, 255, 255, 0.5)' : '#fcfcfd'};
`;
