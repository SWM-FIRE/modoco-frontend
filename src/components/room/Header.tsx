import styled from 'styled-components';
import { useNavigate, useParams } from 'react-router-dom';
import Timer from './Timer';
import Settings from './Settings';
import About from './About';
import { ReactComponent as X } from '../../assets/svg/X.svg';
import { emitLeaveChatRoom } from '../../adapters/chat/socketio';

export default function Header() {
  const navigate = useNavigate();
  const { roomId } = useParams();

  const onClick = () => {
    const result = window.confirm('정말 모도코를 종료하시겠습니까?');
    if (result) {
      emitLeaveChatRoom(roomId);
      navigate('/');
    }
  };

  return (
    <Component>
      <About />
      <Timer />
      <Settings />
      <Button onClick={onClick}>
        나가기 <X />
      </Button>
    </Component>
  );
}

const Component = styled.div`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  /* background-color: rgba(30, 39, 69); */
  background-color: #29292e;
  width: 100%;
  height: 10rem;
  font-family: IBMPlexSansKRRegular;
  color: #f9fafb;
  font-size: 1.8rem;
`;

const Button = styled.button`
  position: absolute;
  display: inline-flex;
  align-items: center;
  justify-content: space-between;
  right: 8.1rem;
  border: 1px solid #94a3b8;
  border-radius: 5rem;
  cursor: pointer;
  color: #f9fafb;
  font-family: IBMPlexSansKRRegular;
  padding: 1rem 2rem;
  font-size: 1.7rem;
  width: 11.7rem;
`;
