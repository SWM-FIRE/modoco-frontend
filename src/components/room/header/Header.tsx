import styled from 'styled-components';
import { useNavigate, useParams } from 'react-router-dom';
import Timer from './Timer';
import Settings from './Settings';
import { ReactComponent as X } from '../../../assets/svg/X.svg';
import Theme from '../Theme';
import roomSocket from '../../../adapters/roomSocket';
import connectedUsersStore from '../../../stores/connectedUsersStore';
import messageStore from '../../../stores/messagesStore';
import { useCreateMediaStream } from '../../../hooks/useCreateMediaStream';
import userPcStore from '../../../stores/userPcStore';

export default function Header({ theme }) {
  const navigate = useNavigate();
  const { roomId } = useParams();
  const { setUsers } = connectedUsersStore();
  const { stopMediaStream } = useCreateMediaStream();
  const { setMessages } = messageStore();
  const { emptyPc } = userPcStore();

  const onClick = () => {
    const result = window.confirm('정말 모도코를 종료하시겠습니까?');
    if (result) {
      roomSocket.emit('leaveRoom', roomId);
      setUsers([]);
      emptyPc();
      setMessages([]);
      stopMediaStream();
      navigate('/');
      window.location.reload();
    }
  };

  return (
    <Component>
      <Theme theme={theme} />
      <Center>
        <Timer />
        <Settings />
      </Center>
      <Button onClick={onClick}>
        나가기 <X />
      </Button>
    </Component>
  );
}

const Center = styled.div`
  display: flex;
`;

const Component = styled.div`
  padding-right: 8rem;
  padding-left: 8rem;
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: ${({ theme }) => theme.header};
  width: 100%;
  height: 10rem;
  font-family: IBMPlexSansKRRegular;
  color: #f9fafb;
  font-size: 1.8rem;
`;

const Button = styled.button`
  display: inline-flex;
  align-items: center;
  border: 1px solid #e5e7eb;
  border-radius: 5rem;
  cursor: pointer;
  color: #f9fafb;
  font-family: IBMPlexSansKRRegular;
  padding: 1rem 2rem;
  font-size: 1.7rem;
  min-width: 11.7rem;
`;
