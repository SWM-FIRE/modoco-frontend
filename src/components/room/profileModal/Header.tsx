import styled from 'styled-components';
import { ReactComponent as X } from '../../../assets/svg/X.svg';

export default function Header({ toggle }: { toggle: (_type) => void }) {
  return (
    <ModalTitle>
      <ExitButton onClick={toggle}>
        <X />
      </ExitButton>
    </ModalTitle>
  );
}

const ExitButton = styled.button`
  cursor: pointer;
`;

const ModalTitle = styled.div`
  display: flex;
  color: #f9fafb;
  margin: 3rem 3rem 0 0;
  font-size: 2rem;
  line-height: 3rem;
  font-family: IBMPlexSansKRRegular;
  align-items: center;
  justify-content: flex-end;
`;
