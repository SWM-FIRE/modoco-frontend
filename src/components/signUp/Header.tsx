import styled from 'styled-components';
import { useNavigate } from 'react-router-dom';
import UserStore from '../../stores/userStore';
import MyAvatar from '../../assets/avatar/MyAvatar';
import LoginModalStore from '../../stores/loginModalStore';

export default function Header() {
  const { nickname, avatar } = UserStore();
  const { openModal } = LoginModalStore();
  const navigate = useNavigate();

  const clickLogo = () => {
    navigate('/#/');
  };

  return (
    <Container>
      <Logo onClick={clickLogo}>modoco</Logo>
      {nickname ? (
        <Profile onClick={openModal}>
          <AvatarContainer>
            <MyAvatar num={Number(avatar)} />
          </AvatarContainer>
          {nickname}
        </Profile>
      ) : (
        <Login onClick={openModal}>로그인</Login>
      )}
    </Container>
  );
}

const Login = styled.button`
  width: 9.1rem;
  height: 4.8rem;
  border: 2px solid #494e5b;
  border-radius: 0.8rem;
  cursor: pointer;
  color: #fcfcf9;
  font-size: 1.6rem;
  font-family: IBMPlexSansKRRegular;
  font-weight: 700;
`;

const Logo = styled.div`
  cursor: pointer;
  font-size: 3.2rem;
  font-family: IBMPlexSansKRRegular, Arial;
  color: white;
  font-weight: 700;
`;

const AvatarContainer = styled.div`
  height: 4rem;
  svg {
    height: 100%;
  }
`;

const Profile = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: IBMPlexSansKRRegular;
  font-size: 1.5rem;
  color: white;
  cursor: pointer;
`;

const Container = styled.header`
  width: 100vw;
  height: 10rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: solid #2b2e41 0.1rem;
  padding: 0 4.4rem;
  background-color: #18181b;
`;
