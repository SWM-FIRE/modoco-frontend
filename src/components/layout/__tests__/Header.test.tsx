import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../Header';

describe('<Header />', () => {
  it('matches snapshot', () => {
    const utils = render(
      <Router>
        <Header />,
      </Router>,
    );
    expect(utils.container).toMatchSnapshot();
  });
  it('has logo', () => {
    const utils = render(
      <Router>
        <Header />,
      </Router>,
    );
    utils.getByText('modoco', { selector: 'h1' });
  });
  it('has loginButton', () => {
    const utils = render(
      <Router>
        <Header />,
      </Router>,
    );
    utils.getByText('로그인');
  });
});
