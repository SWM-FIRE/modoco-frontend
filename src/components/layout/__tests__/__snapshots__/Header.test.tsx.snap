// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<div>
  <div
    class="sc-crXcEl cgWTUl"
  >
    <h1
      class="sc-papXJ dqcJOn"
    >
      modoco
    </h1>
    <button
      class="sc-ftvSup zBHDT"
    >
      로그인
    </button>
  </div>
  ,
</div>
`;
